### PR TITLE
vim: use value variant and update config script

### PR DESCRIPTION
**Pull request by sethrj**
This changes several conflicting variants to a single value variant, and uses conflicts instead of raising InstallError. (With clingo, requesting +gui automatically selects features=huge!)

I have also rearranged the dependencies for clarity and simplified the conifgure args.

### CODE REVIEW

**Review by alalazo — DISMISSED** [review]

**Comment by alalazo** [comment]
Waiting for @haampie review before merging. In case does any of you want to enlist as a maintainer of `vim`?

**Review by haampie — DISMISSED** [review]

**Comment by sethrj** [comment]
Sure, I'll add myself.

**Review by alalazo — CHANGES_REQUESTED** [review]
style issues since we need to conform to #23947

**Comment by sethrj** [comment]
Bleh, so whenever you touch a package now you have to refactor the imports too 🙄 maybe we should have an announcement about that. Also for some reason, `spack style` is *not* picking this up locally for me, even though I have the latest `py-flake8` installed, am rebased off `upstream/develop`, and have my local `develop` up-to-date with upstream/develop.

@adamjstewart  am I missing something?

**Comment by sethrj** [comment]
Whoops shit I also accidentally pushed my openmpi changes to this arggg. Hold on, let me back those out.

**Comment by adamjstewart** [comment]
> so whenever you touch a package now you have to refactor the imports too

Yes, @cosmicexplorer was working on integrating `isort` into `spack style` so that this would be done automatically. Once we drop Python 2.6 support we'll be adding `black` too, which requires things to be formatted very specifically. For now, you'll have to reorder imports manually just like with flake8 things.

> Also for some reason, spack style is not picking this up locally for me

You need to install `flake8-import-order`, it's a separate extension library.

**Comment by sethrj** [comment]
@alalazo @haampie The only change I made was to add myself as maintainer; could I please have a re-review?

**Review by alalazo — APPROVED** [review]

**Comment by sethrj** [comment]
Cycling to try to get a good gitlab run 🙁

**Comment by sethrj** [comment]
Ugh, looks like `hdf5` is borked:
```
==> Error: Pipeline generation failed due to the presence of the following specs that are known to be broken in develop:
507* hdf5@1.10.7%gcc@7.5.0~cxx+fortran+hl~ipo~java+mpi+shared~szip~threadsafe+tools api=default build_type=RelWithDebInfo arch=linux-ubuntu18.04-x86_64 ^autoconf@2.69%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^automake@1.16.3%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^berkeley-db@18.1.40%gcc@7.5.0+cxx~docs+stl patches=b231fcc4d5cff05e5c3a4814f6a5af0e9a966428dc2176540d2c05aff41de522 arch=linux-ubuntu18.04-x86_64 ^cmake@3.20.3%gcc@7.5.0~doc+ncurses+openssl+ownlibs~qt build_type=Release arch=linux-ubuntu18.04-x86_64 ^findutils@4.8.0%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^gdbm@1.19%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^hwloc@2.4.1%gcc@7.5.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml+pci+shared arch=linux-ubuntu18.04-x86_64 ^libfabric@1.12.1%gcc@7.5.0~debug~kdreg fabrics=rxm,sockets,tcp,udp arch=linux-ubuntu18.04-x86_64 ^libiconv@1.16%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^libpciaccess@0.16%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^libsigsegv@2.13%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^libtool@2.4.6%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^libxml2@2.9.10%gcc@7.5.0~python arch=linux-ubuntu18.04-x86_64 ^m4@1.4.19%gcc@7.5.0+sigsegv arch=linux-ubuntu18.04-x86_64 ^mpich@3.4.2%gcc@7.5.0~argobots+fortran+hwloc+hydra+libxml2+pci+romio~slurm~verbs~wrapperrpath device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu18.04-x86_64 ^ncurses@6.2%gcc@7.5.0~symlinks+termlib abi=none arch=linux-ubuntu18.04-x86_64 ^numactl@2.0.14%gcc@7.5.0 patches=4e1d78cbbb85de625bad28705e748856033eaafab92a66dffd383a3d7e00cc94,62fc8a8bf7665a60e8f4c93ebbd535647cebf74198f7afafec4c085a8825c006 arch=linux-ubuntu18.04-x86_64 ^openssl@1.1.1k%gcc@7.5.0~docs+systemcerts arch=linux-ubuntu18.04-x86_64 ^perl@5.34.0%gcc@7.5.0+cpanm+shared+threads arch=linux-ubuntu18.04-x86_64 ^pkgconf@1.7.4%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^readline@8.1%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^util-macros@1.19.1%gcc@7.5.0 arch=linux-ubuntu18.04-x86_64 ^xz@5.2.5%gcc@7.5.0+pic libs=shared,static arch=linux-ubuntu18.04-x86_64 ^zlib@1.2.11%gcc@7.5.0+optimize+pic+shared arch=linux-ubuntu18.04-x86_64 (b5ibryqllolt5abyu35ttofbzhqfuer5)
```

**Pull request merged**
